### ISSUE TITLE
UI does not display error messages when configuring external authentication providers.

### ISSUE THREAD

**Issue by paynejacob** [rancher/dashboard#3841]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. setup an external auth provider (e.g. saml)
2. the saml provider should fail to return the principals

**Result:**

Observe a generic error message on the page when the user should see the error message specified https://github.com/rancher/rancher/blob/master/pkg/auth/providers/saml/saml_client.go#L305

**Expected Result:**

I should see the content of `UITranslationKeyForErrorMessage`

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): dev 113c35317
- Installation option (single install/HA): single, running locally

**Comment by Jono-SUSE-Rancher** [comment]
@gaktive & @nwmac - Hey guys, just putting this on your radar to get looked at as it is a customer facing issue (https://jira.suse.com/browse/SURE-2626) and we're targeting it for 2.6.1.
cc: @SheilaghM

**Comment by davidnuzik** [comment]
Waiting on backend https://github.com/rancher/rancher/pull/33029

**Comment by davidnuzik** [comment]
@Jono-SUSE-Rancher @paynejacob am I correct that this is basically a duplicate of https://github.com/rancher/rancher/issues/13666#issuecomment-946110511 ?
